event-sourcing/event-repository/event-repository-jdbc/src/test/java/uk/gov/justice/services/eventsourcing/repository/jdbc/eventstream/EventStreamJdbcRepositoryIT.java

```diff
@@ -0,0 +1,260 @@
+        final UUID streamId6 = randomUUID();
+        final UUID streamId7 = randomUUID();
+
+        jdbcRepository.insert(eventStreamOf(streamId1, 1L));
+        jdbcRepository.insert(eventStreamOf(streamId2, 2L));
+        jdbcRepository.insert(eventStreamOf(streamId3, 3L));
+        jdbcRepository.insert(eventStreamOf(streamId4, 4L));
+        jdbcRepository.insert(eventStreamOf(streamId5, 5L));
+        jdbcRepository.insert(eventStreamOf(streamId6, 6L));
+        jdbcRepository.insert(eventStreamOf(streamId7, 7L));
+
+        final List<EventStream> streams = jdbcRepository.previous(4L, 3L).collect(toList());
+        assertThat(streams, hasSize(3));
+
+        assertThat(streams.get(0).getStreamId(), is(streamId6));
+        assertThat(streams.get(0).getSequenceNumber(), is(6L));
+
+        assertThat(streams.get(1).getStreamId(), is(streamId5));
+        assertThat(streams.get(1).getSequenceNumber(), is(5L));
+
+        assertThat(streams.get(2).getStreamId(), is(streamId4));
+        assertThat(streams.get(2).getSequenceNumber(), is(4L));
+    }
+
+    @Test
+    public void shouldReturnTrueWhenRecordExists() throws InvalidSequenceIdException {
+        final UUID streamId1 = randomUUID();
+        final UUID streamId2 = randomUUID();
+        final UUID streamId3 = randomUUID();
+        final UUID streamId4 = randomUUID();
+        final UUID streamId5 = randomUUID();
+        final UUID streamId6 = randomUUID();
+        final UUID streamId7 = randomUUID();
+
+        jdbcRepository.insert(eventStreamOf(streamId1, 1L));
+        jdbcRepository.insert(eventStreamOf(streamId2, 2L));
+        jdbcRepository.insert(eventStreamOf(streamId3, 3L));
+        jdbcRepository.insert(eventStreamOf(streamId4, 4L));
+        jdbcRepository.insert(eventStreamOf(streamId5, 5L));
+        jdbcRepository.insert(eventStreamOf(streamId6, 6L));
+        jdbcRepository.insert(eventStreamOf(streamId7, 7L));
+
+
+        final boolean exists = jdbcRepository.recordExists(7);
+        assertTrue(exists);
+    }
+
+    @Test
+    public void shouldReturnFalseWhenRecordDoesNotExist() throws InvalidSequenceIdException {
+        final UUID streamId1 = randomUUID();
+        final UUID streamId2 = randomUUID();
+        final UUID streamId3 = randomUUID();
+        final UUID streamId4 = randomUUID();
+        final UUID streamId5 = randomUUID();
+        final UUID streamId6 = randomUUID();
+        final UUID streamId7 = randomUUID();
+
+        jdbcRepository.insert(eventStreamOf(streamId1, 1L));
+        jdbcRepository.insert(eventStreamOf(streamId2, 2L));
+        jdbcRepository.insert(eventStreamOf(streamId3, 3L));
+        jdbcRepository.insert(eventStreamOf(streamId4, 4L));
+        jdbcRepository.insert(eventStreamOf(streamId5, 5L));
+        jdbcRepository.insert(eventStreamOf(streamId6, 6L));
+        jdbcRepository.insert(eventStreamOf(streamId7, 7L));
+
+        final boolean exists = jdbcRepository.recordExists(10L);
```

Comment:
as above, add 2 streams and then check for the 3rd

Reply:
reduced the no of streams